go/vt/vtgate/planbuilder/operators/rewrite/rewriters.go

```diff
@@ -73,12 +74,22 @@ func TopDown(in ops.Operator, rewriter BreakableFunc) (ops.Operator, error) {
 	return op, err
 }
 
-func bottomUp(root ops.Operator, rewriter Func) (ops.Operator, TreeIdentity, error) {
+func bottomUp(root ops.Operator, rootID semantics.TableSet, resolveID func(ops.Operator) semantics.TableSet, rewriter Func) (ops.Operator, TreeIdentity, error) {
 	oldInputs := root.Inputs()
 	anythingChanged := false
 	newInputs := make([]ops.Operator, len(oldInputs))
+	childID := rootID
+
+	type noLHSTableSet interface{ NoLHSTableSet() }
```

Comment:
Adding a comment that explains why we are introducing this interface might help our future self